src/main/java/com/coders/repository/ProductRepository.java

```diff
@@ -92,13 +64,21 @@ public Map<Integer, Product> getAllProducts() throws NoDataException {
         return productMap;
     }
 
-    public void removeProductByName(String name) throws NotSuchElementException {
-        Product product = productMap.get(name);
+    public Product removeProductByName(String name) throws NotSuchElementException {
```

Comment:
tutaj wyciągasz z  z mapy product o id 0 - jest to błędne.
Napisze tobie  w punktach jak ta metoda powinna wyglądać:
1. stwórz metodę getProductByName(String name) 
- w tej metodzie musisz steamowac po values()
- użyć metody .fliter aby sprawdzac czy produkty streama maja taka nazwe
- .findFirst()

2. jesli produkt nie zostanie znaleziony to rzucasz wyjatek
3. jesli produkt bedzie zwrocony to po jego ID usuwasz z mapy.